### PR TITLE
[SDFAB-980] Update STC scenarios to more closely mimic SMF behavior

### PR DESCRIPTION
**Pull request by Wailoks**
- Perform drop first, then forward in `*-forwarding` scenarios
- Perform drop first, then forward and finally buffer in `pfcp-buffering` scenario
- Set drop flag of downlink FAR during session establishment in `mock-smf`

### CODE REVIEW

**Comment by Wailoks** [comment]
Include changes of https://github.com/omec-project/up4/pull/227

**Comment by Wailoks** [comment]
retest this please

File: scenarios/pfcp-buffering.xml [hunk 1]
```diff
@@ -6,23 +6,58 @@
...
+                <sequential var="${ENODEB#}"
+                            starts="Pfcp-Buffer-Session-Modify-${#}"
+                            ends="Pfcp-Buffer-Session-Modify-${#-1}">
+                    <step name="Pfcp-Buffer-Session-Modify-${#}" requires="Check-Traffic-Buff-Drop-DL" delay="5"
+                          exec="mock-smf-cmd modify --session-count 5 --base ${#}0 --ue-pool 17.0.${#-1}.0/24 --enb-addr 140.0.10${#-1}.1"/>
+                </sequential>
+            </group>
+            <step name="Check-Up4-Flows-Buff-Modify" requires="Pfcp-Push-Buff-Modify" delay="5"
+                  exec="onos-cli-grep ${OCI} up4:read-flows 'UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
+            <step name="Check-Flow-Rules-Buff-Modify" requires="Pfcp-Push-Buff-Modify" delay="5"
```
Comment (pierventre) [inline scenarios/pfcp-buffering.xml:43]: shall we verify the fwd here ?

File: scenarios/pfcp-buffering.xml [hunk 2]
```diff
@@ -6,23 +6,58 @@
...
             </group>
-            <step name="Check-Up4-Flows-Buff" requires="Pfcp-Push-Buff" delay="5"
+            <step name="Check-Up4-Flows-Buff-Buffer" requires="Pfcp-Push-Buff-Buffer" delay="5"
                   exec="onos-cli-grep ${OCI} up4:read-flows 'UL flows=${PFCP_UP4_FLOWS}, DL flows=${PFCP_UP4_FLOWS}'"/>
-            <step name="Check-Flow-Rules-Buff" requires="Pfcp-Push-Buff" delay="5"
+            <step name="Check-Flow-Rules-Buff-Buffer" requires="Pfcp-Push-Buff-Buffer" delay="5"
                   exec="onos-check-flows ${OCI}"/>
         </group>
-        <group name="Check-Traffic-Buff" requires="Pfcp-Set-Buff" delay="5">
+        <group name="Check-Traffic-Buff" requires="Pfcp-Set-Buff-Buffer" delay="5">
```
Comment (pierventre) [inline scenarios/pfcp-buffering.xml:60]: here we should not consider TEIDs being wrong. See my comment on the TEID patch

**Review by pierventre — COMMENTED** [review]
I don't see changes to p4rt-buffering should we modify also that ?

File: scenarios/pfcp-buffering.xml (hunk 1 above)
Comment (daniele-moro) [reply scenarios/pfcp-buffering.xml:43]: No, I believe there is no need to verify fwd here, we already do that in the `pfcp-forwarding` scenario.

**Review by daniele-moro — COMMENTED** [review]
@Wailoks please, update PR name and description.

File: scenarios/p4rt-buffering.xml [hunk 3]
```diff
@@ -32,8 +32,9 @@
                 <step name="Verify-First-Buffer-${#}" requires="^"
                       exec="docker-log-grep ${ODI} 'Received FIRST_BUFFER: ueAddress=17.0.${#-1}.1'"/>
                 <!-- -t 30 to wait enough time for packets to be released after P4rt-Set-Fwd-After-Buff -->
+                <!-- Not going to verify TEID in buffering test -->
```
[the remark this answers is not in the document]
Comment (Wailoks) [reply scenarios/p4rt-buffering.xml:63]: done

**Comment by Wailoks** [comment]
retest this please

**Comment by Wailoks** [comment]
retest this please

**Review by daniele-moro — APPROVED** [review]

**Pull request merged**